todo/src/App.js

```diff
@@ -0,0 +1,87 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { submit_todo, delete_todo, update_completed } from './actions';
+import Todos from './components/Todos';
+import './App.css';
+
+class App extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			todo: {
+				value: '',
+				completed: false,
+				id: '',
+			},
+		};
+	}
+	handleChange = e => {
+		e.preventDefault();
+		this.setState({
+			todo: {
+				...this.state.todo,
+				[e.target.name]: e.target.value,
+			},
+		});
+	};
+
+	handleSubmit = e => {
+		e.preventDefault();
+		const newTodo = {
+			id: Date.now(),
+			value: this.state.todo.value,
+			completed: false,
+		};
+		this.props.submit_todo(newTodo);
+		this.setState({
+			todo: {
+				value: '',
+				completed: false,
+				id: '',
+			},
+		});
+	};
+
+	handleDelete = (e, id) => {
+		e.preventDefault();
+		let filtered = this.props.todos.filter(items => items.id !== id);
+		this.props.delete_todo(filtered);
+	};
+
+	handleCompleted = (e, id) => {
+		e.preventDefault();
+
+		this.props.update_completed(id);
+	};
+
+	render() {
+		return (
+			<div className="App">
+				<form>
+					<input
+						name="value"
+						value={this.state.todo.value}
+						placeholder="Todo"
+						onChange={this.handleChange}
+					/>
+					<button onClick={e => this.handleSubmit(e)}>Submit</button>
+				</form>
+				<Todos
+					handleCompleted={this.handleCompleted}
+					handleDelete={this.handleDelete}
+					todos={this.props.todos}
+				/>
```

Comment:
So, we're getting an error that the console says involves your Todos component here, and I'm not exactly sure where it's coming from. You may check your Todos.js file and see if you can find it. Otherwise, the key warning does not affect how your application actually runs.

> Warning: Each child in an array or iterator should have a unique "key" prop. See https://fb.me/react-warning-keys for more information.
>    in Todos (at App.js:69)